cms/static/js/spec/views/move_xblock_spec.js

```diff
@@ -336,9 +367,226 @@ define(['jquery', 'underscore', 'edx-ui-toolkit/js/utils/spec-helpers/ajax-helpe
+                        clickForwardButton(1);
+                    });
+                    verifyMoveEnabled(true);
+                });
+
+                it('verify move state while navigating', function() {
+                    renderViews(courseOutline, xblockAncestorInfo);
+                    verifyXBlockInfo(courseOutlineOptions, 'section', 0, 'forward', true);
+                    // start from course outline again
+                    modal.moveXBlockBreadcrumbView.$el.find('.bc-container button').first().click();
+                    verifyXBlockInfo(courseOutlineOptions, 'section', 1, 'forward', false);
+                });
+
+                it('moves an xblock when move button is clicked', function() {
+                    var requests = AjaxHelpers.requests(this);
+                    moveXBlockWithSuccess(requests);
+                });
+
+                it('does not move an xblock when cancel button is clicked', function() {
+                    modal.$el.find('.modal-actions .action-cancel').click();
+                    expect(modal.movedAlertView).toBeNull();
+                    expect(getSentRequests().length).toEqual(0);
+                });
+
+                it('undo move an xblock when undo move link is clicked', function() {
+                    var sourceIndex = 0,
+                        requests = AjaxHelpers.requests(this);
+                    moveXBlockWithSuccess(requests);
+                    modal.movedAlertView.$el.find('.action-save').click();
+                    AjaxHelpers.respondWithJson(requests, {
+                        move_source_locator: sourceLocator,
+                        parent_locator: sourceParentLocator,
+                        target_index: sourceIndex
+                    });
+                    verifyUndoConfirmationFeedbackTitle(sourceDisplayName);
+                });
+
+                it('shows a notification when moving', function() {
+                    var requests = AjaxHelpers.requests(this),
+                        notificationSpy = ViewHelpers.createNotificationSpy();
+                    // navigate to a target parent and click
+                    renderViews(courseOutline);
+                    _.each(_.range(3), function() {
+                        clickForwardButton(1);
+                    });
+                    modal.$el.find('.modal-actions .action-move').click();
+                    verifyNotificationStatus(requests, notificationSpy, 'Moving');
+                });
+
+                it('shows a notification when undo moving', function() {
+                    var notificationSpy,
+                        requests = AjaxHelpers.requests(this);
+                    moveXBlockWithSuccess(requests);
+                    notificationSpy = ViewHelpers.createNotificationSpy();
+                    modal.movedAlertView.$el.find('.action-save').click();
+                    verifyNotificationStatus(requests, notificationSpy, 'Undo moving');
```

Comment:
@mushtaqak I'd like to see some tests for failure case. e.g. `Undo Move returned 500`, ` take me there returned 500`, `Actual move returned 500`.